engine/class_modules/paladin/sc_paladin.cpp

```diff
@@ -1467,13 +1468,26 @@ struct divine_toll_t : public paladin_spell_t
 
     if ( result_is_hit( s->result ) )
     {
+      // T31 only procs on the first valid target, others are not affected, even if valid
+      if ( !t31HasProcced && p()->sets->has_set_bonus( PALADIN_RETRIBUTION, T31, B2 ) &&
```

Comment:
You should be able to check whether the current impact is for the primary target via looking at `s->chain_target`

Reply:
Ah - but you can't quite do that here because it'll proc on the "first" expurgation target even if it's not the primary target? Or am I misunderstanding?

Reply:
Correct. Divine Toll shoots up to 5 Judgments and if any of those Judgments hits a target with an Expurgation ticking, it procs the Tier. Even if your current target (chain_target 0) doesn't have Expurgation ticking

Reply:
To clarify further:
Judgment only procs Tier on the main target, even if Blessed Champion is talented and a cleave hits another target with Expurgation
Divine Resonance is exactly the same as a normal Judgment
Divine Toll's initial cast will proc the Tier on the first target it hits that has an Expurgation debuff ticking (including travel time etc. But I don't think it's really needed to simulate this)